### PR TITLE
Features/integrate pypsa eur

### PR DESCRIPTION
**Pull request by ClaraBuettner**
This branch updates the integration of pypsa-eur (which is now used instead of pypsa-eur-sec). 
Changes for pypsa-eur are now done within powerd-data, so we do not need a fork from pypsa-eur anymore. 

## Before merging into `dev`-branch, please make sure that

- [ ] the `CHANGELOG.rst` was updated.
- [ ] new and adjusted code is formated using `black` and `isort`.
- [ ] the `Dataset`-version is updated when existing datasets are adjusted.
- [ ] the branch was merged into the
      `continuous-integration/run-everything-over-the-weekend`-branch.
- [ ] the workflow is running successful in `test mode`.
- [ ] the workflow is running successful in `Everything` mode.

### CODE REVIEW

File: src/egon/data/datasets/pypsaeur/__init__.py [hunk 1]
```diff
@@ -3,117 +3,99 @@
...
-import importlib_resources as resources
 import numpy as np
 import pandas as pd
 import pypsa
+import shutil
 import yaml
 
 from egon.data import __path__, db, logger, config
 from egon.data.datasets import Dataset
+from egon.data.datasets.pypsaeur.manipulate_prepared_network import execute
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/pypsaeur/__init__.py]: @ulfmueller This line runs into errors

File: src/egon/data/datasets/pypsaeur/manipulate_prepared_network.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        "Generator",
+        network.generators[
+            network.generators.carrier.str.contains("rural solar thermal")
+        ].index,
+    )
+
+    network.links.loc["DE1 0 services rural ground heat pump", "p_nom_min"] = 0
+
+
+def execute():
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/pypsaeur/manipulate_prepared_network.py]: @ulfmueller. The function is defined here

**Comment by CarlosEpia** [comment]
I installed powerd-data using this branch and started a new calculation for the status2019 scenario. Below you can find relevant information:
-airflow-port: 8081
-database-name: powerd-data
-database-port: '59731'
- working directory: home/powerd/powerd-run-de
- virtual environment: home/powerd/powerd-run-de/venv
- git repository: home/powerd/powerd-run-de/powerd-data

**Comment by CarlosEpia** [comment]
Airflow starts but the same error is displayed for both pipelines:

Broken DAG: [/home/powerd/powerd-run-de/powerd-data/src/egon/data/airflow/dags/pipeline_status_quo.py]
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/operators/python.py", line 56, in <module>
    from airflow.utils.file import get_unique_dag_module_name
ImportError: cannot import name 'get_unique_dag_module_name' from 'airflow.utils.file' (/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/utils/file.py)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/powerd/powerd-run-de/powerd-data/src/egon/data/datasets/__init__.py", line 12, in <module>
    from airflow.operators.python_operator import PythonOperator
  File "/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/utils/deprecation_tools.py", line 63, in getattr_with_deprecation
    raise ImportError(error_message) from e
ImportError: Could not import `airflow.operators.python.PythonOperator` while trying to import `airflow.operators.python_operator.PythonOperator`.

**Comment by ClaraBuettner** [comment]
> Airflow starts but the same error is displayed for both pipelines:
> 
> Broken DAG: [/home/powerd/powerd-run-de/powerd-data/src/egon/data/airflow/dags/pipeline_status_quo.py] Traceback (most recent call last): File "", line 219, in _call_with_frames_removed File "/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/operators/python.py", line 56, in from airflow.utils.file import get_unique_dag_module_name ImportError: cannot import name 'get_unique_dag_module_name' from 'airflow.utils.file' (/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/utils/file.py)
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last): File "/home/powerd/powerd-run-de/powerd-data/src/egon/data/datasets/**init**.py", line 12, in from airflow.operators.python_operator import PythonOperator File "/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/utils/deprecation_tools.py", line 63, in getattr_with_deprecation raise ImportError(error_message) from e ImportError: Could not import `airflow.operators.python.PythonOperator` while trying to import `airflow.operators.python_operator.PythonOperator`.

Thanks for reporting! I can somehow not reproduce the problem locally. Does the environment still exist on the server?

**Comment by ClaraBuettner** [comment]
Since the problem looked like #218, I merged that branch.
If it solves the problem, I will merge the other PR.

**Comment by CarlosEpia** [comment]
> > Airflow starts but the same error is displayed for both pipelines:
> > Broken DAG: [/home/powerd/powerd-run-de/powerd-data/src/egon/data/airflow/dags/pipeline_status_quo.py] Traceback (most recent call last): File "", line 219, in _call_with_frames_removed File "/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/operators/python.py", line 56, in from airflow.utils.file import get_unique_dag_module_name ImportError: cannot import name 'get_unique_dag_module_name' from 'airflow.utils.file' (/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/utils/file.py)
> > The above exception was the direct cause of the following exception:
> > Traceback (most recent call last): File "/home/powerd/powerd-run-de/powerd-data/src/egon/data/datasets/**init**.py", line 12, in from airflow.operators.python_operator import PythonOperator File "/home/powerd/powerd-run-de/venv/lib/python3.8/site-packages/airflow/utils/deprecation_tools.py", line 63, in getattr_with_deprecation raise ImportError(error_message) from e ImportError: Could not import `airflow.operators.python.PythonOperator` while trying to import `airflow.operators.python_operator.PythonOperator`.
> 
> Thanks for reporting! I can somehow not reproduce the problem locally. Does the environment still exist on the server?

Sure. Everything related to this test is still available in at31:
-airflow-port: 8081
-database-name: powerd-data
-database-port: '59731'

working directory: home/powerd/powerd-run-de
virtual environment: home/powerd/powerd-run-de/venv
git repository: home/powerd/powerd-run-de/powerd-data

**Comment by ClaraBuettner** [comment]
@CarlosEpia: I am very optimistic that pypsa-eur will finish now successfully. There were some problems with downloading files (inside pypsa-eur). I tried to fix this by copying some files, which is not the best solution (I'm sorry!).
 In general, the download works fine when I test it locally, so it might be related to the server. 

I run pypsa-eur outside of airflow to speed up the bug fixing. But I think you can now either restart the task "pypsa_eur.prepare_network" or just mark it as successful as it was already running in a separate process. 
At least it is possible now to continue running the pipeline. I wish I would have been able to do a complete run before I leave. But I hope it also works out like this and you can continue from this point on.

**Comment by CarlosEpia** [comment]
There are several references in the function `execute` to files that are not available:
- "data_bundle_powerd_data/district_heating_shares_egon.csv"
- "input-pypsa-eur-sec/heat_demand_timeseries_DE_eGon100RE.csv"
- "input-pypsa-eur-sec/electrical_demand_timeseries_DE_eGon100RE.csv"
- "input-pypsa-eur-sec/geothermal_potential_germany.csv"

File: src/egon/data/datasets/pypsaeur/__init__.py [hunk 3]
```diff
@@ -1160,28 +1161,263 @@ def overwrite_H2_pipeline_share():
...
+            network.generators.carrier.str.contains("rural solar thermal")
+        ].index,
+    )
+
+    network.links.loc["DE1 0 services rural ground heat pump", "p_nom_min"] = 0
+
+
+def execute():
+    with open(
+        __path__[0] + "/datasets/pypsaeursec/config.yaml", "r"
```
Comment (CarlosEpia) [inline src/egon/data/datasets/pypsaeur/__init__.py]: Do you mean `__path__[0] + "/datasets/pypsaeur/config.yaml", "r"` ?

File: src/egon/data/datasets/pypsaeur/__init__.py [hunk 4]
```diff
@@ -3,117 +3,101 @@
...
-            [
-                "git",
-                "clone",
-                "--branch",
-                "v0.3.0",
-                "https://github.com/PyPSA/technology-data.git",
-                technology_data_repos,
-            ]
+        # Copy config file for egon-data to pypsa-eur directory
+        shutil.copy(
```
Comment (CarlosEpia) [inline src/egon/data/datasets/pypsaeur/__init__.py]: There is a problem with this command. I adjusted it in the commit 70e27dd and then it worked.

**Review by CarlosEpia — CHANGES_REQUESTED** [review]
At the end, it was not possible to run these tasks locally or on the server. I also tried to use the pre-network already generated but the next tasks failed as well.

**Comment by CarlosEpia** [comment]
The tasks included in the new class `RunPypsaEur` must be included as a dependency for `ElectricalNeighbours`

**Comment by ClaraBuettner** [comment]
Some of the missing files are created when eGon100RE is one of the selected scenarios. One file needs to be added to the data bundle and in one other case, I need to have a look at the code. 
But at least it works when the missing files are added manually.

**Comment by ClaraBuettner** [comment]
> The tasks included in the new class `RunPypsaEur` must be included as a dependency for `ElectricalNeighbours`

That is strange, the prepared network should be enough. That is already part of the dependencies. Do you remember which error you got?

**Comment by CarlosEpia** [comment]
I copy pasted the file Snakefile from your session in order to get over this problem: Error: Snakefile "run-pypsa-eur/Snakefile" not found.; 331359)

**Comment by ClaraBuettner** [comment]
This PR was replaced by the (already merged) PR #293

**Pull request closed without merging**